lib/user.rb

```diff
@@ -0,0 +1,22 @@
+class User
+# attr_reader :name
+attr_accessor :jokes, :name
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(name, joke)
```

Comment:
A better name for `name` here is `user`, because it reminds us of the type of argument we are passing in, which is a `User` object. Then, based on the interaction pattern, we can see that we should expect the `User` object that is passed in as an argument to receive the `Joke` object that is also passed in as the second argument.